### PR TITLE
`tmpnet`: Move tmpnet/local to tmpnet package 

### PR DESCRIPTION
**Pull request by marun**
Previously the tmpnet code was separated into `tmpnet` and `tmpnet.local` packages. The intention was for `tmpnet` to define common interfaces and configuration, and `tmpnet.local` to orchestrate process-based temporary networks.

I eventually realized that the previous abstractions were unnecessary. A process- or kube-based network need only differ in its runtime - the code that starts/stops/etc the node. The configuration will be largely similar and can all be stored in the same way. Hence this PR: removing the concept of `Local{Node,Network,Config}` in favor of Node/Network/Config. Subsequent PRs will introduce the concept of `NodeRuntime` to support the choice of nodes run as processes or pods.

### CODE REVIEW

File: tests/upgrade/upgrade_test.go [hunk 1]
```diff
@@ -48,12 +48,11 @@ var _ = ginkgo.Describe("[Upgrade]", func() {
 	require := require.New(ginkgo.GinkgoT())
 
 	ginkgo.It("can upgrade versions", func() {
-		// TODO(marun) How many nodes should the target network have to best validate upgrade?
```
Comment (abi87) [inline tests/upgrade/upgrade_test.go:51]: q: why we dropped the TODO?

File: tests/fixture/tmpnet/config.go [hunk 2]
```diff
@@ -206,6 +197,24 @@ func (nc *NodeConfig) EnsureKeys() error {
 	return nc.EnsureNodeID()
 }
 
+// Derives the nodes proof-of-possession. Requires the node to have a
+// BLS signing key.
+func (nc *NodeConfig) GetProofOfPosession() (*signer.ProofOfPossession, error) {
```
Comment (abi87) [inline tests/fixture/tmpnet/config.go:202]: typo: GetProofOfPossession?

File: tests/fixture/tmpnet/config.go [hunk 3]
```diff
@@ -49,14 +49,13 @@ var (
...
-	errNoKeysForGenesis            = errors.New("failed to generate genesis: no keys to fund")
-	errInvalidNetworkIDForGenesis  = errors.New("network ID can't be mainnet, testnet or local network ID")
-	errMissingValidatorsForGenesis = errors.New("no genesis validators provided")
-	errMissingBalancesForGenesis   = errors.New("no genesis balances given")
-	errMissingTLSKeyForNodeID      = fmt.Errorf("failed to ensure node ID: missing value for %q", config.StakingTLSKeyContentKey)
-	errMissingCertForNodeID        = fmt.Errorf("failed to ensure node ID: missing value for %q", config.StakingCertContentKey)
-	errInvalidKeypair              = fmt.Errorf("%q and %q must be provided together or not at all", config.StakingTLSKeyContentKey, config.StakingCertContentKey)
+	errNoKeysForGenesis           = errors.New("failed to generate genesis: no keys to fund")
+	errInvalidNetworkIDForGenesis = errors.New("network ID can't be mainnet, testnet or local network ID")
+	errMissingStakersForGenesis   = errors.New("no genesis stakers provided")
```
Comment (abi87) [inline tests/fixture/tmpnet/config.go:54]: q: do we set in genesis stakers that are not validators?

File: tests/fixture/tmpnet/config.go [hunk 4]
```diff
@@ -409,25 +419,5 @@ func NewTestGenesis(
 	}
 	config.CChainGenesis = string(cChainGenesisBytes)
 
-	// Give staking rewards for initial validators to a random address. Any testing of staking rewards
```
Comment (abi87) [inline tests/fixture/tmpnet/config.go:412]: q: why we dropped this block?

File: tests/fixture/tmpnet/network.go [hunk 5]
```diff
@@ -81,39 +82,25 @@ func FindNextNetworkID(rootDir string) (uint32, string, error) {
...
 }
 
-// Defines the configuration required for a local network (i.e. one composed of local processes).
-type LocalNetwork struct {
-	tmpnet.NetworkConfig
-	LocalConfig
+// Defines the configuration required for a tempoary network
+type Network struct {
+	NetworkConfig
+	NodeRuntimeConfig
```
Comment (abi87) [inline tests/fixture/tmpnet/network.go:88]: q: is is the same runtime config for every node in the network?

**Review by abi87 — COMMENTED** [review]
mostly questions and nits

File: tests/e2e/c/dynamic_fees.go [hunk 6]
```diff
@@ -42,14 +42,14 @@ var _ = e2e.DescribeCChain("[Dynamic Fees]", func() {
...
+		key := privateNetwork.PreFundedKeys[0]
 		ethAddress := evm.GetEthAddress(key)
 
 		ginkgo.By("initializing a coreth client")
-		node := privateNetwork.GetNodes()[0]
+		node := privateNetwork.Nodes[0]
 		nodeURI := tmpnet.NodeURI{
-			NodeID: node.GetID(),
-			URI:    node.GetProcessContext().URI,
+			ID:  node.ID,
```
Comment (aaronbuchwald) [inline tests/e2e/c/dynamic_fees.go:51]: optional nit: I think using NodeID is more explicit than ID, which can be confusing at times, so I'd prefer to keep the field name `NodeID` here

**Review by aaronbuchwald — APPROVED** [review]
LGTM - one optional nit based on my own preference (NodeID instead of ID)

**Review by darioush — APPROVED** [review]

**Pull request merged**